src/main/java/net/neoforged/neoforge/client/pipeline/PipelineModifier.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.pipeline;
+
+import com.mojang.blaze3d.pipeline.RenderPipeline;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion;
+
+/**
```

Comment:
The javadoc should link to the event and to the render system methods that can apply the modifier.